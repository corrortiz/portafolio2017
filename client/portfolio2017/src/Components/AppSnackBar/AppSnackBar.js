import React, { Component } from 'react';
import PropTypes from 'prop-types';
//MUI Components
import Button from 'material-ui/Button';
import Snackbar from 'material-ui/Snackbar';
import Slide from 'material-ui/transitions/Slide';
import { withStyles } from 'material-ui/styles';
//HOC for connect to Redux
import GlobalsConnect from '../../HOC/GlobalsConnect/GlobalsConnect';
//locale neds of global connect to work
import { lenguajeSelector } from '../../Store/Actions/globals';
import { Close } from '../../Assets/diccionary';

//Func for deside te type of transiton to the sbackbar
function TypeOfTransition(props) {
  return <Slide direction="up" {...props} />;
}
//CSS in JS style
const styles = theme => ({
  snackbar: {
    margin: theme.spacing.unit
  }
});

/**
 * SnackBar component with auto hide and animation
 */
export class AppSnackBar extends Component {
  state = {
    transition: TypeOfTransition
  };

  handleClick = transition => () => {
    this.props.showSnackBar();
    this.setState({ transition });
  };

  handleClose = () => {
    this.props.showSnackBar();
  };

  render() {
    const { transition } = this.state;
    const { openSnackBar, messageSnackBar, lenguaje } = this.props.globals;
    const { classes } = this.props;

    return (
      <div className="footer__appSnackBar">
        {/*This button is necesary for adding the custom transition and is hidden with css*/}
        <Button className="hide" onClick={this.handleClick(TypeOfTransition)}>
          Down
        </Button>

        <Snackbar
          open={openSnackBar}
          onClose={this.handleClose}
          onRequestClose={this.handleClose}
          autoHideDuration={3500}
          transition={transition}
          className={classes.snackbar}
          SnackbarContentProps={{
            'aria-describedby': 'message-id'
          }}
          message={<span id="message-id">{messageSnackBar}</span>}
          action={[
            <Button
              key="close"
              aria-label="Close"
              color="accent"
              dense
              onClick={this.handleClose}
            >
              {lenguajeSelector(lenguaje, Close)}
            </Button>
          ]}
        />
      </div>
    );
  }
}

AppSnackBar.propTypes = {
  globals: PropTypes.shape({
    openSnackBar: PropTypes.bool.isRequired,
    messageSnackBar: PropTypes.string.isRequired,
    lenguaje: PropTypes.string.isRequired
  })
};

AppSnackBar = withStyles(styles)(AppSnackBar);

export default GlobalsConnect(AppSnackBar);
