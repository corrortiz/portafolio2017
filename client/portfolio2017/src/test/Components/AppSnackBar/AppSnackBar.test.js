import React from 'react';
import { shallow } from 'enzyme';
import { AppSnackBar } from '../../../Components/AppSnackBar/AppSnackBar';

const globals = {
  openSnackBar: true,
  messageSnackBar: 'AO HyS Rules',
  lenguaje: 'es'
};

describe('AppSnackBar Component', () => {
  it('renders without crashing', () => {
    shallow(<AppSnackBar globals={globals} />);
  });

  it('should compare a snapshot of AppSnackBar', () => {
    const wrapper = shallow(<AppSnackBar globals={globals} />);
    expect(wrapper).toMatchSnapshot();
  });
});
