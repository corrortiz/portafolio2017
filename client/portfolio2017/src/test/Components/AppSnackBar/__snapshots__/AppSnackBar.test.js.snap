// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AppSnackBar Component should compare a snapshot of AppSnackBar 1`] = `
<AppSnackBar
  classes={
    Object {
      "snackbar": "AppSnackBar-snackbar-1",
    }
  }
  globals={
    Object {
      "lenguaje": "es",
      "messageSnackBar": "AO HyS Rules",
      "openSnackBar": true,
    }
  }
/>
`;
